src/Controllers/API.php

```diff
@@ -352,35 +128,22 @@ protected function renameResource($data)
 
         Versioned::set_stage($currentStage);
 
-        return $this->json(true, 200, 1);
+        return Helper::json(true, 200, 1);
     }
 
     /**
-     * @param array $data
-     * @param int $status
-     * @param int $ttl
-     * @return HTTPResponse
+     * @param string $publicId
+     * @param string $resourceType
+     * @return array
      */
-    protected function json($data, $status = 200, $ttl = 300)
+    protected function getResourceData($publicId, $resourceType)
     {
-        $response = new HTTPResponse();
+        $resource = Helper::get_resource($publicId, $resourceType);
 
-        $response->setBody(json_encode($data));
+        $return = Helper::process_resource($resource);
 
-        $response->setStatusCode($status);
+        $this->extend('updateResourceData', $return, $resource);
 
-        $expire = new DateTime();
-        $expire->add(new DateInterval('PT' . $ttl . 'S'));
-        $expire->setTimezone(new DateTimeZone('UTC'));
-
-        $response->addHeader('Content-Type', 'application/json; charset=utf-8')
-            ->addHeader('Access-Control-Allow-Methods', 'GET')
-            ->addHeader('Access-Control-Allow-Credentials', 'true')
-            ->addHeader('Access-Control-Allow-Origin', '*')
-            ->addHeader('Cache-Control', 'public, must-revalidate, stale-while-revalidate=86400, no-transform')
-            ->addHeader('Expires', $expire->format('D, d M Y H:i:00 \G\M\T'))
-            ->addHeader('Vary', 'Origin');
-
-        return $response;
+        return $return;
     }
```

Comment:
If you use the `SilverStripe\Core\Extensible` trait in the Helper then you can add the `::extend()` call in there instead of doing it this way

Reply:
Can't call the extend method on a static.